### ISSUE TITLE
Verification loop, can't log in

### ISSUE THREAD

**Issue by mateuszpokojowczyk** [instabot-py/instabot.py#905]
- [x] I have searched for other issues with the same problem.

## Hosting
- [x] Are you hosting the bot on a VPS or other Cloud hosting?

#### Select one:
- [x] Bug

### Operating System? (include version)
Linux

## Python Version?
3

## Description

Login error, I'm stuck in verification loop.
I keep clicking "it was me", also has been trying to verify from the same ip.
I use private proxy with my account and it has been working well before I was trying to restart the bot.

**Comment by ugurozturk** [comment]
@mateuszpokojowczyk it is probabbly nothing about the repo. Perhaps the private proxy as your said is already blocked ip on instagram side so you can't log in via that

**Comment by Nicolas-vrcc** [comment]
@ugurozturk 
There is a lot a people having this issue and it's not related to a blocked IP or anything because it works 1 day but the other it stops and I get stuck in the verification loop. I hope someone will fix this.

**Comment by ugurozturk** [comment]
@Nicolas-vrcc test https://github.com/instabot-py/instabot.py/tree/fake_user_agent

**Comment by Nicolas-vrcc** [comment]
Ok, so I've done a lot of exepriments and here is what I found out:

- I can use the script with any account as long as I run it from my local terminal ( even account that belong to friends and who've never been related to my local IP ) -> **Works Fine**

- However : Launching it from a VPS or by using a proxy results in a 'suspicious login' endless loop.
-> **Fail**

I don't think it's a user_agent problem because no matter what I try it doesn't change anything...

**Comment by gersanchez** [comment]
the same for me! :(

**Comment by ugurozturk** [comment]
@Nicolas-vrcc how many account are you running ? at the same time same ip ?

**Comment by Nicolas-vrcc** [comment]
@ugurozturk 1 account

**Comment by unformatt** [comment]
My guess is that you're running the script from a known service (i.e. set of IP addresses) that is blocked by Instagram because they do not want bots.

**Comment by Nicolas-vrcc** [comment]
@unformatt Thanks you for your answer. Do you know how to change this on a VPS ?

**Comment by unformatt** [comment]
Try using a host that lesser known than e.g. aws, linode, digital ocean, slice host etc...

**Comment by spaez18** [comment]
I'm having the same issue, when i try to run the bot it says "Can't Login! Connection error!".

I was running the bot on my local computer and also tried running it on a AWS server, but none of them worked for me. Early morning they were running good, but now they keep showing me that error.

**Comment by mactrash** [comment]
@ugurozturk fake-user-agent not work here too

**Comment by luciatseng** [comment]
What worked for me was to disable two-factor authentication on your Instagram account.

**Comment by mactrash** [comment]
I never use two factor authentication and so it is not related.

**Comment by marcinex** [comment]
I have the same problem, bot works about 3 months without any problems and now I have this loop. I can't also login with my account data directly at instagram.com which is really strange.

**Comment by marcinex** [comment]
Ok - solution - change password to new one at instagram.com. It helps me :)

**Comment by Nicolas-vrcc** [comment]
@ugurozturk This really sucks... Hope someone will find an answer to this. I'm searching right now

**Comment by kemong** [comment]
This is happen to me too.
I have around 8 accounts or more, let's say A, B, C, D...
I have 2 VPS. 1 PC and 1Phone.
All 8 accounts can log into Instagram via Android Apps.
But.
When i log in using bot program in my PC, account A can't log in. BUT, when i open up a browser, and login via Instagram web for account A, everything was just fine.
I also get suspicious login attempt warning on Instagram web. Which is kinda hilarious, i use same IP bot browser and bot program, but Instagram only block log in process via bot program, log in from browser was all just fine.
Damn weird.

**Comment by kemong** [comment]
And also, I got not liked 400 for account B on VPS 1, got login error on VPS 2, BUT everything is fine when i run bot program on my PC. 
This makes me wonder, that we should make a script to handle suspicious login attempt warning when log in via bot. To click on "THIS WAS ME" button on that warning.

**Comment by mactrash** [comment]
@kemong maybe instagram.com set some kind of cookie to the browser, sso it work with real browser just guess...

It is because their detect is spot on... just blocked the bot.

**Comment by kemong** [comment]
@mactrash To make it more hilarious, let's take a look on this picture 
![text4427](https://user-images.githubusercontent.com/22748697/35114835-cb35045c-fcb8-11e7-9a98-dd1872468d56.png)

There, i have 3 terminals open. A and B are bots for account kemong.
And C is bot for account mactrash. (Let's use that name anyway..)

Bot A and B running simultaneously, no problem. A for mode 3 and B for mode 1.
Before i took this picture, there were 4 terminals with C running mode 3 and D running for mode 1.
Both C and D were running for Mactrash account.

For some reason, due to exception bot on terminal D stop running. So i have to restart it using "python mactrash.py". Suddenly, on terminal D, bot can't log in. I then open Instagram Apps, as usual, the app told me that someone suspicious was trying to log in to mactrash account.
And then the bot C get exception, can't scan photos on recent feed. It still logged in but can't do anything due to continuous exception.

UNTIL i hit "this was me" button on instagram app, Bot C can continue to work well.
I tried to log in to terminal D using mactrash account, but then it was getting log in error all the time. 
Weird.

I predict that if i close terminal C and bot for mactrash stop running, then i restart the bot, it'll just can't log in just like what was happened on the terminal D.

Clearly, Instagram knows my IP. It blocks Mactrash from log in and also Kemong account to login in another terminal. They're just still running because of a bug in Instagram site. It's just like those bots are already in, when they go out or stop working, i can't make them go to Instagram server anymore using bot program.
But it's not blocking my IP. I still can login to those accounts using App or Browser. 

I THINK INSTAGRAM IS NO LONGER BLOCKS ACCOUNTS, because they will lose big numbers of users if they do it. Once your account is marked suspicious, you can't use bot program on another VPS.
I don't have any idea to deal with it, my only solution for now is only destroy old Droplet (VPS) and start a new one, then i run bots on that fresh new VPS.

But again, i have another account which can't login on any fresh VPS using bot. Only Instagram app or Browser.

**Comment by mactrash** [comment]
@kemong new ip and just run it once it will keep going. DON'T QUIT / RESTART the bot! It what I face now.

**Comment by mactrash** [comment]
@kemong couple real activities from the phone app seems lead to unblock the ip... need more people to confirm.

**Comment by Nicolas-vrcc** [comment]
Do you guys think getting a private proxy and running the script through it could be the solution ?

**Comment by mactrash** [comment]
@Nicolas-vrcc I run it on a home use ADSL with just one account it also act like that (can't login if launch say... after 3 times) so I think a private proxy may not help.

**Comment by Nicolas-vrcc** [comment]
Does this mean that this script is dying :'( ?

**Comment by mactrash** [comment]
@Nicolas-vrcc config a general setting (will no change at all , lower like lower follow) and run it in background in a fresh ip, or may be try the mobile real activities as I mention before see is that that ip unblock... and run it in the background... I have now run couple hrs. and instagram have block my likes once, then I click report issue in my mobile and now is another couple hrs. without issue.

**Comment by ugurozturk** [comment]
fake-user-agent updated. it does store the first time created user-agent then it will use it for future usage.

**Comment by ugurozturk** [comment]
@mactrash @Nicolas-vrcc as a suprise here we go. https://github.com/instabot-py/instabot.py/tree/save_cookies it does store the user login and it wont try to login again. I need tests for it. Perhaps it can fix your "suspicious login" . Note : use it on mode 0. I only made changes for mode 0. if it works fine we can implement for other modes. Yet i made a lot changes that i am not sure if it is really oke for this repo like removed ig_ keys. And dont know what gonna happen after a failure login. Because everything did run just fine here.

Note : if everything goes wrong remove `follows_db.dbcookies.txt` file

**Comment by mactrash** [comment]
@ugurozturk sorry they block my account, so I Cannot test now.

**Comment by mactrash** [comment]
@ugurozturk test already... it never get success login, there much be some act from the script that different to a normal access and make instagram bot it at once?

**Comment by ugurozturk** [comment]
@mactrash i can't fix if i can't see :)

**Comment by mactrash** [comment]
@ugurozturk no problem, anyway thanks for your work many thanks.

**Comment by hardcash** [comment]
I have a suggestion that unfortunately I don't have the knowledge to implement. I have my boot hosted on aws and noticed I cannot login on the bot, but I am able to login on the browser under same IP after clicking "it was me". I saw @ugurozturk made a branch to save cookies to the bot. I was wondering if there's a way to import cookies from the browser to bypass the suspicious login.

**Comment by mactrash** [comment]
@ugurozturk @hardcash chrome have an 3rd party extension name cookies.txt that can export cookies, is that possible to work?

I try to export and use as the follows_db.dbcookies.txt file directly, but not working maybe the format is different.

**Comment by ugurozturk** [comment]
@hardcash there is a python function MozillaCookieJar. But it says it has a lot of buggies. If you want to go run and learn some python check that.  @mactrash yes the format looks so different.

**Comment by mactrash** [comment]
@ugurozturk can you post a sample of follows_db.dbcookies.txt that save by instabot.py so I may try manual match the file I capture from chrome and try if that work.

**Comment by Nicolas-vrcc** [comment]
@ugurozturk @mactrash 
Alright, it stopped working for 2 days on my VPS and it worked again for 1 day and then stopped again for another 2 days on all of my accounts...
What the actual heck ? I am 100% sure that the error is IP related and has nothing to do with the user-agent because I tried everything with it ( changing it ect...) but it still doesn't work...

**Comment by mactrash** [comment]
@hardcash even it is success login but there is no follows_db.dbcookies.txt saved.

**Comment by may47** [comment]
@ugurozturk  Hey tried running the save_cookies branch getting the following message pls help!
![capture](https://user-images.githubusercontent.com/9697636/35220867-01aa6c1a-ff9e-11e7-9c1e-0224bea3ec05.PNG)

**Comment by ugurozturk** [comment]
@may47 Try again, updated it now

**Comment by ugurozturk** [comment]
@Nicolas-vrcc Tell us what did you change and didn't work. So we can tell others that change is not working. Perhaps your limits are too high ? what is your limits i re checked but i realized that i didn't see it

**Comment by ugurozturk** [comment]
@mactrash are you sure about you are using save_cookies branch ?

**Comment by mactrash** [comment]
@ugurozturk ok last msg is my fault I clone the source and think will be the save_cookies branch

Now download the zip, and success login and saved the cookie.txt and then CTRL C quite the bot and restart the bot then result error

traceback (most recent call last):
  File "example888.py", line 53, in <module>
    unfollow_whitelist=['example_user_1', 'example_user_2'])
  File "/home/x/instabot.py-save_cookies/src/instabot.py", line 233, in __init__
    self.login()
  File "/home/x/instabot.py-save_cookies/src/instabot.py", line 282, in login
    x_csrftoken = cj._cookies['.www.instagram.com']['/']['csrftoken'].value
KeyError: '.www.instagram.com'

**Comment by ugurozturk** [comment]
@mactrash Yeah the part looks branch bug. Yet i need your help here to fix that Because i modifies as i get the cookies however it looks like you are having different cookies. Probabbly `['.www.instagram.com']` this part different.

Can you write here like
`Set-Cookie3: csrftoken=xxxx; path="/"; domain=".www.instagram.com"; path_spec; `
Do control and **dont paste here your csrftoken** !

**Comment by mactrash** [comment]
@ugurozturk 

#LWP-Cookies-2.0
Set-Cookie3: csrftoken=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx; path="/"; domain="www.instagram.com"; path_spec; secure; expires="2019-01-21 13:37:22Z"; version=0
Set-Cookie3: ds_user_id=xxxxxxxxxx; path="/"; domain="www.instagram.com"; path_spec; expires="2018-04-22 13:37:22Z"; version=0
Set-Cookie3: mid=xxxxxxxxxxxxxxxxxxxxxxxxxxxx; path="/"; domain="www.instagram.com"; path_spec; expires="2038-01-17 13:37:18Z"; version=0
Set-Cookie3: sessionid="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"; path="/"; domain="www.instagram.com"; path_spec; secure; expires="2018-04-22 13:37:22Z"; HttpOnly=None; version=0

**Comment by ugurozturk** [comment]
@mactrash so it comes to you like `www.instagram.com` but me `.www.instagram.com` To temp fix for you is change instabot.py:282 to `x_csrftoken = cj._cookies['www.instagram.com']['/']['csrftoken'].value` instead `x_csrftoken = cj._cookies['.www.instagram.com']['/']['csrftoken'].value`

**Comment by ugurozturk** [comment]
@mactrash also updated the branch. So if you can't edit by yourself, just update your code from latest commit

**Comment by mactrash** [comment]
@ugurozturk ok it run but...

22.01.2018_23:58 Removing already liked medias..
22.01.2018_23:58 Trying to unfollow #1: 
22.01.2018_23:58 Removing already liked medias..
22.01.2018_23:58 Removing already liked medias..
22.01.2018_23:58 Removing already liked medias..
22.01.2018_23:58 Removing already liked medias..

Just like it cannot login without login error

**Comment by ugurozturk** [comment]
@mactrash it seems like it didn't login. remove the cookie file and try again

**Comment by mactrash** [comment]
@ugurozturk remove the cookie file it do login and run normally (lucky day), but then quit it and restart.. the bot should use the cookie there and skip login ... but it act like cannot login

**Comment by aopiskin** [comment]
@mactrash Probaly yes.

**Comment by mactrash** [comment]
@aopiskin @ugurozturk wait 10min+ run again still same 

23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Trying to unfollow #1: 
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..
23.01.2018_01:04 Removing already liked medias..

Except del the cookie.txt file, otherwise noway to run again.

**Comment by may47** [comment]
@aopiskin  How are you able to run 3 accounts parallely in the same folder pls. help!!
@ugurozturk  Great work brother you ultimately found a way and your work has really helped a lot of us!

**Comment by ugurozturk** [comment]
@may47 create a file for each of the accounts like "example.py". Change username password and database_name paramater values. It is done.

**Comment by aopiskin** [comment]
@may47 as a matter of fact, I creat a few files as said @ugurozturk ex. "account1.ph" "account2.ph" "account3.ph" and I run all off them with terminal. That sit. 

But still i need to learn how work exactly this sh*t :D @ugurozturk

**Comment by may47** [comment]
@ugurozturk @aopiskin  Thanks a lot brothers!!!

**Comment by ugurozturk** [comment]
@aopiskin @may47 dont forget about database_name. Otherwise all of your accounts will use same database.

**Comment by gersanchez** [comment]
@ugurozturk @may47 Really thanks for your contributions, right now is working nice!
I have a little question, I'm running 3 accounts parallelly but I don't understand that stuff about the database_name parameter: what I have to change, where I find "database_name", what is that?

Thanks so much!

**Comment by ugurozturk** [comment]
@gersanchez it is a parameter just like username or password or taglist etc. check this https://github.com/instabot-py/instabot.py/blob/master/README.md . Use unique database_name each of your username

**Comment by gersanchez** [comment]
yup, but. for example.
account1.py and account2.py where is related to which "database_name"
I know there is a "follows_db.db" but how I know with which account is related.

**Comment by ugurozturk** [comment]
@gersanchez https://github.com/instabot-py/instabot.py/issues/905#issuecomment-360207222 As i said there. It is parameter and your giving it.

**Comment by digital-bloom** [comment]
#980

**Comment by ChipWolf** [comment]
If this is still an issue, please open a new issue. Thanks!
